### ISSUE TITLE
Error: Argument 1 of predicate color expected something of sort $int but got something of sort $i

### ISSUE THREAD

**Issue by stanmoon** [vprover/vampire#333]
The following code produces a parsing error:
`
tff(def1, type, color: $i * $int > $o).

tff(axiom1, axiom, ![T:$int] : color(green,T) => color(green,T)).

tff(axiom2, axiom, color(green,0)).

tff(query,conjecture, color(green,5)).
`

The error message:

`
Parsing Error on line 3
Argument 1 of predicate color expected something of sort $int but got something of sort $i
`

It took a while to figure out that the actual problem is that quantifiers require parentheses.
Adding them solves the problem:

`
tff(def1, type, color: $i * $int > $o).

tff(axiom1, axiom, ![T:$int] : (color(green,T) => color(green,T))).

tff(axiom2, axiom, color(green,0)).

tff(query,conjecture, color(green,5)).
`

`
% Refutation not found, incomplete strategy% ------------------------------
% Version: Vampire 4.5.1 (commit 57a6f78c on 2020-07-15 11:59:04 +0200)
% Termination reason: Refutation not found, incomplete strategy

% Memory used [KB]: 4861
% Time elapsed: 0.040 s
% ------------------------------
% ------------------------------
`
The issue is to improve the error message in this case.

Thanks,

**Comment by selig** [comment]
The main warning here is that in the first problem (without parentheses) T is not under a quantifier. This might be enough information to give a more informative error message. I'll take a look.

**Comment by selig** [comment]
Sorry, I just realised that the fix from your last issue (#322)  should apply here i.e. I get the following on the first program

```
% Parsing Error on line 3
Failed to create predicate application for color of type ($i * $int) > $o
The sort $i of the intended term argument X0 (at index 1) is not an instance of sort $int
```

This might be extendible to checking whether X0 is quantified with sort $int or not.
